api/core/v1alpha1/clustercatalog_types.go

```diff
@@ -85,153 +115,269 @@ type ClusterCatalogSpec struct {
 	//    image:
 	//      ref: quay.io/operatorhubio/catalog:latest
 	//
-	// For more information on FBC, see https://olm.operatorframework.io/docs/reference/file-based-catalogs/#docs
+	// +kubebuilder:validation:Required
 	Source CatalogSource `json:"source"`
 
-	// priority is an optional field that allows the user to define a priority for a ClusterCatalog.
+	// priority allows the user to define a priority for a ClusterCatalog.
+	// priority is optional.
+	//
 	// A ClusterCatalog's priority is used by clients as a tie-breaker between ClusterCatalogs that meet the client's requirements.
+	// It is up to clients to decide how to handle scenarios where multiple ClusterCatalogs with the same priority meet their requirements.
 	// For example, in the case where multiple ClusterCatalogs provide the same bundle.
-	// A higher number means higher priority. Negative numbers are also accepted.
-	// When omitted, the default priority is 0.
+	// A higher number means higher priority.
+	//
+	// When omitted, the default priority is 0 because that is the zero value of integers.
+	//
+	// Negative numbers can be used to specify a priority lower than the default.
+	// Positive numbers can be used to specify a priority higher than the default.
+	//
+	// The lowest possible value is -2147483648.
+	// The highest possible value is 214748647.
+	//
 	// +kubebuilder:default:=0
+	// +kubebuilder:validation:minimum:=-2147483648
+	// +kubebuilder:validation:maximum:=214748647
 	// +optional
 	Priority int32 `json:"priority"`
 
-	// Availability is an optional field that allows users to define whether the ClusterCatalog is utilized by the operator-controller.
+	// availabilityMode allows users to define how the ClusterCatalog is made available to clients on the cluster.
+	// availabilityMode is optional.
 	//
-	// Allowed values are : ["Enabled", "Disabled"].
-	// If set to "Enabled", the catalog will be used for updates, serving contents, and package installations.
+	// Allowed values are "Available" and "Unavailable" and omitted.
```

Comment:
This sorta makes it sound like "empty string" is one of the valid values. Maybe we just say that the allowed values are "Available" and "Unavailable", and that it will be defaulted to "Available" if unset by the client. This gets nuanced with server-side apply, so just want to make sure we are careful about how we word this.

```suggestion
	// Allowed values are "Available" and "Unavailable".
```

Reply:
Joel had mentioned that for optional fields that are defaulted when omitted that it is typical to include "omitted" as a valid value and below explain what happens when the value is omitted

Reply:
Joel's comment was on the ClusterExtension API, but I figured it applied here as well: https://github.com/openshift/api/pull/2067#discussion_r1812459328